packages/cli/src/cmd/serve.js

```diff
@@ -111,8 +137,9 @@ function serve(userSpecifiedRoot, options) {
       const server = liveServer.start(serverConfig);
       server.addListener('listening', () => {
         const address = server.address();
-        const serveHost = address.address === '0.0.0.0' ? '127.0.0.1' : address.address;
-        const serveURL = `http://${serveHost}:${address.port}`;
+        const serveHost = address.address;
```

Comment:
What do you think about adding some validation to the user input host address?

I believe that this could be better for UX as
- it checks the host address before building the pages (which could take some time)
- it better handles the errors that are being thrown. Currently users will be exposed to errors like `Error: getaddrinfo ENOTFOUND 127.0.300.1`

I was thinking some sort of simple validation like checking if the input is an actual IP address and if any number in the address exceeds 255

Reply:
Hi, good idea! I was thinking if we want to do this validation, we can actually just do the validations (including check if port is unavailable, check if host is being used etc) all at once before building the pages. Currently, all these validations are done after building the pages. 

Since this PR is focusing on adding another small feature, I suggest we create a separate issue and pull request to specifically address this validation issue. What do you think?

Reply:
Yep I think a separate issue is fine as well! I believe there is some level of error handling for when the host is being used. From my experience if I specified a used host, the nearest unused hostname will be used, which allows the user to still serve the site. I think we can discuss about what needs to be validated once the issue is opened!